config/crds/vsphereproviderconfig_v1alpha1_vspheremachineproviderconfig.yaml

```diff
@@ -0,0 +1,103 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  creationTimestamp: null
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: vspheremachineproviderconfigs.vsphereproviderconfig.sigs.k8s.io
+spec:
+  group: vsphereproviderconfig.sigs.k8s.io
+  names:
+    kind: VsphereMachineProviderConfig
+    plural: vspheremachineproviderconfigs
+  scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        machineRef:
+          type: string
+        machineSpec:
+          properties:
+            datacenter:
+              type: string
+            datastore:
+              type: string
+            disks:
+              items:
+                properties:
+                  diskLabel:
+                    type: string
+                  diskSizeGB:
+                    format: int64
+                    type: integer
+                type: object
+              type: array
+            memoryMB:
+              format: int64
+              type: integer
+            networks:
+              items:
+                properties:
+                  ipConfig:
+                    properties:
+                      dns:
+                        items:
+                          type: string
+                        type: array
+                      gateway:
+                        type: string
+                      ip:
+                        type: string
+                      netmask:
+                        type: string
+                      networkType:
+                        type: string
+                    required:
+                    - networkType
+                    type: object
+                  networkName:
+                    type: string
+                required:
+                - networkName
+                type: object
+              type: array
+            numCPUs:
+              format: int32
+              type: integer
+            preloaded:
+              type: boolean
+            resourcePool:
+              type: string
+            template:
+              type: string
+            vmFolder:
+              type: string
+            vsphereCloudInit:
+              type: boolean
+          required:
+          - datacenter
+          - datastore
+          - networks
+          - template
+          - disks
+          type: object
+        machineVariables:
+          type: object
+        metadata:
+          type: object
+        vsphereMachine:
+          type: string
+      required:
+      - vsphereMachine
+      - machineVariables
```

Comment:
Can we make this optional to begin with in this structure here?
We can remove this completely once we remove the terraform related code all together

Reply:
we'll just remove all the terraform code in another PR soon.